### PR TITLE
Keep track of last visited on date

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Add last_visited_on column to users table
- Update last_visited_on when requesting HTML page

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -1164,6 +1164,7 @@
     t.datetime "updated_at", null: false
     t.boolean "show_on_supporters_page", default: true, null: false
     t.datetime "disabled_at"
+    t.date "last_visited_on"
```
Comment (ErikSchierboom) [inline db/schema.rb:1167]: Should this be indexed?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@ class ApplicationController < ActionController::Base
   before_action :set_request_context
   after_action :set_body_class_header
   after_action :set_csp_header
+  after_action :updated_last_visited_on!
```
Comment (ErikSchierboom) [inline app/controllers/application_controller.rb:14]: I don't know if you want this to be a `before_action` 🤷

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -132,6 +133,14 @@ def store_user_location!
     store_location_for(:user, request.fullpath)
   end
 
+  def updated_last_visited_on!
+    return unless user_signed_in?
+    return unless request.format == 'html'
```
Comment (ErikSchierboom) [inline app/controllers/application_controller.rb]: Not sure if this is the "proper" way to do it, but the aim was to not updated the visited on date when doing an API request

File: db/schema.rb (hunk 1 above)
Comment (iHiD) [reply db/schema.rb:1167]: Yeah, probably.

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (iHiD) [reply app/controllers/application_controller.rb]: I think request.format is a symbol, not a string (based on `config/initializers/devise.rb`)

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (iHiD) [reply app/controllers/application_controller.rb:14]: This is fine.

**Comment by iHiD** [comment]
Needs rebasing and being green :)

**Review by iHiD — APPROVED** [review]

**Pull request merged**